app/forms/updateRegisteredDetails/ChangeRegisteredDetailsFormProvider.scala

```diff
@@ -0,0 +1,32 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package forms.updateRegisteredDetails
+
+import javax.inject.Inject
+
+import forms.mappings.Mappings
+import play.api.data.Form
+import play.api.data.Forms.set
+import models.updateRegisteredDetails.ChangeRegisteredDetails
+
+class ChangeRegisteredDetailsFormProvider @Inject() extends Mappings {
+
+  def apply(): Form[Set[ChangeRegisteredDetails]] =
```

Comment:
I think we need some form validation to check the isVoluntary flag as I think (its hard to read) the old code validated in the form
i..e
-  when im voluntary, i am allowed these checkboxes
- when im not voluntary, im not allowed to submit x checkbox

(taken from L300 in Variations journey in old fe
  ```  import ContactChangeType._
    val changeOptions =
      if (data.isVoluntary)
        ContactChangeType.values.filter(_ != ContactChangeType.Sites)
      else
        ContactChangeType.values

    for {
      change <- ask[Set[ContactChangeType]](
                 "change-registered-details",
                 validation = Rule.nonEmpty[Set[ContactChangeType]] alongWith Subset(changeOptions: _*)
               )```